### DISCUSSION TITLE
workflow queries

### DISCUSSION THREAD

**Discussion by hahahah-single** [GreenleafLab/ArchR#932] [Questions / Documentation]
Hi ,after Harmony reduction when I run addImputeWeights which reducedDims should I use "IterativeLSI" or "Harmony", and if I should reuse "addClusters" based on IterativeLSI  reducedDims to calculate  cluster for downstream group coverage generation and call peak, or "Harmony" is also useful.

**Comment by rcorces** [comment]
I'm not sure if this is a matter of opinion but my feeling is that it is safer to perform `addImputeWeights` on `IterativeLSI` reduced dims than on Harmony. However, `addClusters` should definitely be performed on `Harmony` and not on `IterativeLSI`. This isnt perfectly showcased in the manual because we dont really use the Harmony-corrected reduction for much on that dataset.

**Reply by hahahah-single** [reply]
Thanks so after harmony reduction we should use harmony reducedDims to calculate cluster for downstream group coverage generation and call peak？but i also see the question in https://github.com/GreenleafLab/ArchR/issues/519 you may more recommend using IterativeLSI space for downstream analysis. I'm confused

**Reply by rcorces** [reply]
the advice given in #519 is the same as what I gave you. After running Harmony, you call clusters and perform downstream cluster-related operations on the Harmony reduced dims space.

**Reply by plbngl** [reply]
Hello thank you for the great work. I have a related question so I am asking it here if that's ok.
I used Harmony to remove sample batch effect and worked well. What would you recommend to run the addGeneIntegrationMatrix() [using an existing seurat object of matched scRNA samples] and addPeak2GeneLinks() functions on: reducedDims = "IterativeLSI" or reducedDims = "Harmony"? 
Thank you so much!
Paola

**Reply by rcorces** [reply]
You should follow the recommendations in the manual.

**Reply by plbngl** [reply]
Oh my bad, I really did not see that in the manual. Would you mind pointing me to the page where this is specified if that's not a problem ? Thanks!!!

**Reply by paupaiz** [reply]
Hi Paola, If you created a reducedDims object named “Harmony” then you should pass that one instead of the iterativeLSI object to any functions where you want to account for batch correction. 
Ref: https://www.archrproject.com/bookdown/batch-effect-correction-wtih-harmony.html

**Reply by YiweiNiu** [reply]
Hi @rcorces,

> it is safer to perform addImputeWeights on IterativeLSI reduced dims than on Harmony

Could you elaborate a bit on this point? If Harmony has a better batch correction, in theory it has a better neighbor representation for imputation, right? Thanks!
